packages/react/src/reactrouterv6-compat-utils.tsx

```diff
@@ -94,14 +94,19 @@ export function createV6CompatibleWrapCreateBrowserRouter<
     }
 
     router.subscribe((state: RouterState) => {
-      const location = state.location;
       if (state.historyAction === 'PUSH' || state.historyAction === 'POP') {
-        handleNavigation({
-          location,
-          routes,
-          navigationType: state.historyAction,
-          version,
-          basename,
+        // Use requestAnimationFrame to wait for Suspense boundaries to settle
+        requestAnimationFrame(() => {
```

Comment:
this will _slightly_ skew all timestamps, even if there is no suspense etc. happening, right? As this will realistically add ~20ms or so before `handleNavigation()` is called.

Not a blocker IMHO, but something to consider. Can we make this smarter (e.g. know when this is lazy?) somehow, possibly...?